pkg/operator/workload_controller_openshiftapiserver_v311_00.go

```diff
@@ -89,49 +89,95 @@ func syncOpenShiftAPIServer_v311_00_to_latest(c OpenShiftAPIServerOperator, orig
 
 	// only manage the apiservices if we have ready pods for the daemonset.  This makes sure that if we're taking over for
 	// something else, we don't stomp their apiservices until ours have a reasonable chance at working.
-	if actualDaemonSet != nil && actualDaemonSet.Status.NumberReady > 0 {
-		err = manageAPIServices_v311_00_to_latest(c.apiregistrationv1Client)
+	var actualAPIServices []*apiregistrationv1.APIService
+	if actualDaemonSet != nil && actualDaemonSet.Status.NumberAvailable > 0 {
+		actualAPIServices, err = manageAPIServices_v311_00_to_latest(c.apiregistrationv1Client)
 		if err != nil {
 			errors = append(errors, fmt.Errorf("%q: %v", "apiservices", err))
 		}
 	}
 
 	// manage status
-	if actualDaemonSet.Status.NumberAvailable > 0 {
+	var availableConditionReason string
+	var availableConditionMessages []string
+	availableConditionStatus := operatorv1.ConditionFalse
+
+	switch {
+	case actualDaemonSet == nil:
+		availableConditionStatus = operatorv1.ConditionUnknown
+		availableConditionReason = "DaemonAvailabilityUnknown"
+		availableConditionMessages = append(availableConditionMessages, "daemonset/apiserver.openshift-apiserver: status unknown")
+	case actualDaemonSet.Status.NumberAvailable > 0 && len(actualAPIServices) == 0:
```

Comment:
make these two different dimensions, same as you did for the two progresing conditions

Reply:
The test `actualDaemonSet.Status.NumberAvailable > 0` lets us know that we actually attempted to manage the `apiServices`